packages/core/src/router.ts

```diff
@@ -1,13 +1,13 @@
 import type { Dictionary } from '@crawlee/types';
 
-import type { CrawlingContext } from './crawlers/crawler_commons';
+import type { RestrictedCrawlingContext } from './crawlers/crawler_commons';
 import { MissingRouteError } from './errors';
 import type { Request } from './request';
 import type { Awaitable } from './typedefs';
 
 const defaultRoute = Symbol('default-route');
 
-export interface RouterHandler<Context extends CrawlingContext = CrawlingContext> extends Router<Context> {
+export interface RouterHandler<Context extends Omit<RestrictedCrawlingContext, 'enqueueLinks'> = RestrictedCrawlingContext> extends Router<Context> {
```

Comment:
I'm not very happy about this. Maybe the type bound should be something else entirely?

Reply:
Isn't this breaking? I guess for people who use `Router.create()` it could be (as you no longer have the `enqueueLinks` helper in the context I think?), although they should be using `create*Router` factories instead.

I would maybe remove those changes and only handle the fact that `enqueueLinks` are not there for basic crawler only in `createBasicRouter()`, that's the only place where the change is actually doing something.

But it's fine to keep it this way too I guess, polishing those things could end up breaking and its better to keep that for v4 (where we also want to get rid of the `Record` type extension).

Reply:
Right, the default value should not change, astute observation! If we get that out of the way, relaxing the bound on the type parameter should not break anything, right?

Reply:
Yeah makes sense